server/src/lib.rs

```diff
@@ -491,12 +533,50 @@ fn config_location(id: u32) -> ObjectStorePath {
     path
 }
 
+// base location in object store for a given database name
+fn database_object_store_path(writer_id: u32, database_name: &DatabaseName<'_>) -> ObjectStorePath {
+    let mut path = ObjectStorePath::default();
+    path.push(format!("{}", writer_id));
+    path.push(database_name.to_string());
+    path
+}
+
+const STORE_ERROR_PAUSE_SECONDS: u64 = 100;
+
+/// Spawns a tokio task that will continuously try to persist the bytes to the
+/// given object store location.
+fn persist_bytes(data: Bytes, store: Arc<ObjectStore>, location: ObjectStorePath) {
```

Comment:
```suggestion
fn persist_bytes_in_background(data: Bytes, store: Arc<ObjectStore>, location: ObjectStorePath) {
```

I got confused when reading the `handle_replicated_write` above -- it looked like the code was not going to acknowledge the write until the segment was persisted. Which is not the case.

Reply:
Renamed in 9d953849166ab3addea5a14bb2ca45db87e23cf9